extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/security/SecurityEventUtil.java

```diff
@@ -0,0 +1,136 @@
+package io.quarkus.opentelemetry.runtime.tracing.security;
+
+import static io.quarkus.security.spi.runtime.AuthenticationFailureEvent.AUTHENTICATION_FAILURE_KEY;
+import static io.quarkus.security.spi.runtime.AuthorizationFailureEvent.AUTHORIZATION_FAILURE_KEY;
+
+import java.time.Instant;
+import java.util.function.BiConsumer;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.quarkus.arc.Arc;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.spi.runtime.AuthenticationFailureEvent;
+import io.quarkus.security.spi.runtime.AuthenticationSuccessEvent;
+import io.quarkus.security.spi.runtime.AuthorizationFailureEvent;
+import io.quarkus.security.spi.runtime.AuthorizationSuccessEvent;
+import io.quarkus.security.spi.runtime.SecurityEvent;
+
+/**
+ * Synthetic CDI observers for various {@link SecurityEvent} types configured during the build time use this util class
+ * to export the events as the OpenTelemetry Span events.
+ */
+public final class SecurityEventUtil {
+    public static final String QUARKUS_SECURITY_NAMESPACE = "quarkus.security.";
+    public static final String AUTHN_SUCCESS_EVENT_NAME = QUARKUS_SECURITY_NAMESPACE + "authentication.success";
+    public static final String AUTHN_FAILURE_EVENT_NAME = QUARKUS_SECURITY_NAMESPACE + "authentication.failure";
+    public static final String AUTHZ_SUCCESS_EVENT_NAME = QUARKUS_SECURITY_NAMESPACE + "authorization.success";
+    public static final String AUTHZ_FAILURE_EVENT_NAME = QUARKUS_SECURITY_NAMESPACE + "authorization.failure";
+    public static final String OTHER_EVENT_NAME = QUARKUS_SECURITY_NAMESPACE + "other";
+    public static final String SECURITY_IDENTITY_PRINCIPAL = QUARKUS_SECURITY_NAMESPACE + "identity.principal";
+    public static final String SECURITY_IDENTITY_IS_ANONYMOUS = QUARKUS_SECURITY_NAMESPACE + "identity.anonymous";
+    public static final String QUARKUS_SECURITY_OTHER_EVENTS_NAMESPACE = QUARKUS_SECURITY_NAMESPACE + "other.";
+    public static final String FAILURE_NAME = QUARKUS_SECURITY_NAMESPACE + "failure.name";
+    public static final String AUTHORIZATION_CONTEXT = QUARKUS_SECURITY_NAMESPACE + "authorization.context";
+
+    private SecurityEventUtil() {
+        // UTIL CLASS
+    }
+
+    /**
+     * Adds {@link SecurityEvent} as Span event. Called from synthetic method observer.
+     */
+    public static void addAllEvents(SecurityEvent event) {
```

Comment:
Can you please add a renaming warning on the public methods of this class?... Because the method names must be aligned with what's usid in the TracingProcessor.